classes/converter.php

```diff
@@ -89,6 +101,87 @@ final public function convert_moodle_url_to_pdf(moodle_url $url, string $filenam
         }
     }
 
+    /**
+     * Convert the given moodle URLs to PDF and store them in the file system.
+     * Note: If the currently logged in user does not have the correct capabilities to view the
+     * target URL, the created PDF will most likely be an error page.
+     *
+     * @param array $urls An array of moodle_url objects representing the target URLs to convert.
+     * @param string $filename the name to give converted file.
+     * (if none is specified, filename will be generated {@see \tool_pdfpages\helper::get_moodle_url_pdf_filename})
+     * @param array $options any additional options to pass to converter, valid options vary with converter
+     * instance, see relevant converter for further details.
+     * @param bool $keepsession should session be maintained after conversion? (For security reasons, this should always be `false`
+     * when conducting a conversion outside of a browser window, such as in an adhoc task or other background process, to prevent
+     * session hijacking.)
+     * @param string $cookiename cookie name to apply to conversion (optional).
+     * @param string $cookievalue cookie value to apply to conversion (optional).
+     * @param array $renderoptions Chromium-specific options:
+     *        - windowSize: An array specifying the size of the browser window, e.g. [1920, 1080].
+     *        - userAgent: A string representing the custom user agent to use when navigating the page.
+     *        - jsCondition: A JavaScript condition to be evaluated, specified as a string.
+     *                It should return a boolean value indicating whether the condition has been met.
+     *        - jsConditionParams: An array of parameters to pass to the Javascript function.
+     *        Note: These options are ignored when using wkhtmltopdf.
+     * @param bool $printpagenumbers Whether to print page numbers in the footer of the combined PDF (optional,
+     * defaults to false).
+     *
+     * @return \stored_file the stored file created during conversion.
+     */
+    final public function convert_moodle_urls_to_pdf(array $urls, string $filename = '', array $options = [],
+            bool $keepsession = false, string $cookiename = '', string $cookievalue = '', array $renderoptions = [],
+            bool $printpagenumbers = false): \stored_file {
+        global $USER, $PAGE;
+
+        $allurlsarevalid = empty(array_filter($urls, fn($url) => !$url instanceof moodle_url));
```

Comment:
Let's make it `fn($url): bool =>...`

Reply:
Added fixup commit https://github.com/catalyst/moodle-tool_pdfpages/pull/40/commits/40f56da8d76a7e2fc0dbc4c47087e57b70bee9ab